lib/krikri/enrichments/language_to_lexvo.rb

```diff
@@ -0,0 +1,153 @@
+    #   lang.providedLabel
+    #   #=> ['finnish']
+    #   lang.exactMatch.map(&:to_term)
+    #   #=> [#<RDF::Vocabulary::Term:0x9b URI:http://lexvo.org/id/iso639-3/fin>]
+    #
+    # @example with no match
+    #   lang = enrich_value('moomin')
+    #   #=> #<DPLA::MAP::Controlled::Language:0x3f(default)>
+    #   lang.providedLabel
+    #   #=> ['moomin']
+    #   lang.exactMatch
+    #   #=> []
+    #
+    # @param value [ActiveTriples::Resource, #to_s]
+    # @return [DPLA::MAP::Controlled::Language, nil] a resource representing the
+    #   language match.
+    def enrich_value(value)
+      return enrich_node(value) if value.is_a?(ActiveTriples::Resource) &&
+        value.node?
+      return value if value.is_a?(DPLA::MAP::Controlled::Language)
+      return nil if value.is_a?(ActiveTriples::Resource)
+      enrich_literal(value)
+    end
+
+    ##
+    # Runs the enrichment over a specific node, accepting an
+    # `ActiveTriples::Resource` with a provided label and returning a new node
+    # with a lexvo match.
+    #
+    # @param value [ActiveTriples::Resource] a resource with a
+    #   `dpla:providedLabel`
+    # @return [Array<ActiveTriples::Resource>, ActiveTriples::Resource]
+    def enrich_node(value)
+      labels = value.get_values(RDF::DPLA.providedLabel)
+      return value if labels.empty?
+      labels.map { |label| enrich_literal(label) }
+    end
+
+    ##
+    # Runs the enrichment over a string.
+    #
+    # @param label [#to_s] the string to match
+    # @return [ActiveTriples::Resource] a blank node with a `dpla:providedLabel`
+    #   of `label` and a `skos:exactMatch` of the matching lexvo language,
+    #   if any
+    def enrich_literal(label)
+      return nil unless label.to_s
+      node = DPLA::MAP::Controlled::Language.new()
+      node.providedLabel = label
+      node.exactMatch = match_iso(label.to_s) || match_label(label.to_s)
+      node
+    end
+
+    ##
+    # Converts string or symbol for a three letter language code to an
+    # `ActiveTriples::Resource`.
+    #
+    # @param code [#to_sym] a three letter iso code
+    # @return [DPLA::MAP::Controlled::Language]
+    def match_iso(code)
+      match = QNAMES.find { |c| c == code.to_sym }
+      from_sym(match)
+    end
+
+    ##
+    # Converts string or symbol for a language label to an
+    # `ActiveTriples::Resource`.
+    #
+    # @param code [#to_sym] a string to match against a language label
+    # @return [DPLA::MAP::Controlled::Language]
+    def match_label(label)
+      @lang_cache ||= {}
+      return @lang_cache[label] if @lang_cache.keys.include? label
+
+      match = TERMS.find do |t|
+        Array(t.label).map(&:downcase).include? label.downcase
+      end
+
+      @lang_cache[label] = from_sym(match)
+    end
+
+    private
+
+    ##
+    # @param code [#to_s] A language code to convert to a URI
+    # @return [DPLA::MAP::Controlled::Language]
+    def from_sym(code)
+      return nil if code.nil?
```

Comment:
> must be empty in some sense, or you would want to be able to evaluate it as empty or false.

Yeah, it would be "empty" in the sense that it's a `AT::Resource` with `RDF::Node` rather than an `RDF::URI`.  I don't think that much helps us re: passing the object around unless, as at L109, we're aiming to return a `Node`.